src/client/remoteinputhandler.cpp

```diff
@@ -0,0 +1,100 @@
+#include "remoteinputhandler.h"
+#include "client/keycode.h"
+#include "hud.h"
+#include "irr_v2d.h"
+#include "remoteclient.capnp.h"
+
+void RemoteInputHandler::step(float dtime) {
+  // skip first loop, because we don't have an observation yet
+  // as draw is called after step
+  if (isFirstLoop) {
+    isFirstLoop = false;
+    return;
+  }
+
+  // send current observation
+  irr::video::IVideoDriver *driver = m_rendering_engine->get_video_driver();
+  irr::video::IImage *image;
+  if (m_rendering_engine->headless) {
+    image = m_rendering_engine->get_screenshot();
+  } else {
+    image = driver->createScreenShot(video::ECF_R8G8B8);
+  }
+
+  ::capnp::MallocMessageBuilder obs_prot;
+  Observation::Builder obs_builder = obs_prot.initRoot<Observation>();
+
+  // parse reward from hud
+  // during game startup, the hud is not yet initialized, so there'll be no
+  // reward for the first 1-2 steps
+  assert(m_player && "Player is null");
+  for (u32 i = 0; i < m_player->maxHudId(); ++i) {
+    auto hudElement = m_player->getHud(i);
```

Comment:
hud_element

Reply:
bit of a lost cause, codebase is super inconsistent re naming conventions